--- tsclient-2.0.2/src/plugins/default/tsc-xdmcp-connection.c.orig	2011-08-26 17:24:02.000000000 +0530
+++ tsclient-2.0.2/src/plugins/default/tsc-xdmcp-connection.c	2011-08-26 17:25:06.000000000 +0530
@@ -138,7 +138,7 @@
 		tsc_spawn_append_args (connection->spawn, "-geometry", geometry, NULL);
 	}
 
-	tsc_spawn_append_args (connection->spawn, "-query", connection->host);
+	tsc_spawn_append_args (connection->spawn, "-query", connection->host, NULL);
 	tsc_spawn_append_args (connection->spawn, display, NULL);
 	g_free (display);
 	g_free (geometry);
